### PR TITLE
Create routes for specific profile details

### PR DESCRIPTION
**Pull request by Eakanath1**
- Updated `userModel`
  - Renamed `workTitle` to `tagLine`
  - Added property `location` for user
  - Updated `maxlength` for few properties
- Updated `userRoutes`
  - Made several `GET` routes `public` including `/api/users/profile`
  - Added new routes for `basicinformation`, `skills`, `languages`, `education`, `experience`, `certification`, `awardsandachievements`, `projects`
- Updated `user` controller
  - Made **get**, **update** and **delete**( for few properties) for the routes mentioned earlier
  - For **get** methods *user_id* was taken using query params instead of request user object, thus enabling it to be a public route

### CODE REVIEW

File: backend/routes/userRoutes.js [hunk 1]
```diff
@@ -27,15 +47,59 @@ router.get("/getteams",protect,getTeamById);
 
 router.post("/login", login);
 router
-  .route("/profile")
-  .get(protect, getUserProfile)
-  .put(protect, updateUserProfile);
+  .route("/profile/")
+  .get(getUserProfile);
```
Comment (veerreshr) [inline backend/routes/userRoutes.js:51]: May be remove getUserProfile and replace it with getUserProfileByUsername function. Dont change the route name. 
And Remove this function
```
router.get(
  "/getuserprofile",
  checkQuery("username").not().isEmpty().trim().escape(),
  getUserProfileByUsername
);
```
 if its not there do a rebase with main. you can refer https://github.com/veerreshr/team-maker/blob/main/CONTRIBUTING-gitflow.md for rebasing

File: backend/controllers/user.js [hunk 2]
```diff
@@ -5,17 +5,17 @@ import Team from "./../models/teamModel.js";
 import mongoose from "mongoose";
 
 // @desc    Get user profile
-// @route   Get /api/users/profile
-// @access  Private
+// @route   Get /api/users/profile?_id=123456
+// @access  Public
 const getUserProfile = expressAsyncHandler(async (req, res) => {
```
Comment (veerreshr) [inline backend/controllers/user.js:10]: you can remove this function completely as we already have `getUserProfileByUsername` function for this purpose

File: .vscode/settings.json [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (veerreshr) [inline .vscode/settings.json:1]: If this is not intentional, remove this file from pushing, add to gitignore and delete it

**Review by veerreshr — CHANGES_REQUESTED** [review]

**Comment by Eakanath1** [comment]
Files are modified as per the request, check them now.

**Pull request merged**